test/unit-tests/reporter-handlers.test.js

```diff
@@ -0,0 +1,77 @@
+const sinon = require('sinon');
+const expect = require('chai').expect;
+const { EventEmitter } = require('events');
+
+// mock event emitter to simulate newman and socket-client
+const mockNewman = new EventEmitter();
+const mockSocket = new EventEmitter();
+
+const ioStub = sinon.stub().callsFake(() => {
+    return mockSocket;
+});
+
+// io() is a default export of socket.io-client; resolve it to export
+// a sinon stub instead
+require.cache[require.resolve('socket.io-client')] = {
+    exports: ioStub,
+};
+
+// require the reporter only after io() has been stubbed
+const reporter = require('../../reporter/index');
+
+describe('Reporter event handlers', () => {
+    before(() => {
+        reporter(mockNewman);
+    });
+
+    it('should connect to the broker', () => {
```

Comment:
I would suggest using different tests for each handler. You can test multiple cases for the handlers based on inputs.

Reply:
I would refactor this in the pub/sub architecture PR